src/ion/data_structures.rs

```diff
@@ -417,7 +417,7 @@ pub struct Env<'a, F: Function> {
     // was to the approprate PReg.
     pub multi_fixed_reg_fixups: Vec<MultiFixedRegFixup>,
 
-    pub inserted_moves: Vec<InsertedMove>,
+    pub inserted_moves: [Vec<InsertedMove>; InsertMovePrio::Count as usize],
```

Comment:
I like using the [enum-map](https://docs.rs/enum-map/) crate for this sort of thing, I find that it makes the code much cleaner.

Reply:
enum-map uses a proc macro and unsafe code. Cranelift currently doesn't depend on any proc macros.

Reply:
That definitely looks like a nicer api than what I'm doing here! As depending on `enum-map` would require us to audit it as a dependency in cranelift, I'm inclined to stick with the current approach. Perhaps using a well-placed `debug_assert` would make the implementation cleaner?

Reply:
@elliottt are you thinking an assert that the actual priority is < `Count`, or...? (That seems somewhat unnecessary to me but perhaps I'm not seeing somewhere else an assert would make sense)

Re: `enum-map`, I'll note that in addition to the concerns above, we try to be really careful about build time here (this is the reason why deps and proc macros in particular are avoided). I like that we have fairly lean crates low in the dependency graph and want to try to keep it that way :-)